tests/unit/rate_limits/test_rate_limit.py

```diff
@@ -157,9 +160,58 @@ def test_determine_if_entity_is_rate_limited_false(self):
             == False
         )
 
+    def test_determine_if_entity_is_rate_limited_error(self, mocker):
+        mock_redis = MagicMock(name="fake_redis")
+        mock_redis.exists.side_effect = RedisError
+        assert (
+            determine_if_entity_is_rate_limited(
+                redis_connection=self.redis_connection,
+                key_name="random_non_existent_key",
+            )
+            == False
+        )
+
     def test_set_entity_to_rate_limited_success(self):
         key_name = "owner_id_123"
         set_entity_to_rate_limited(
             redis_connection=self.redis_connection, key_name=key_name, ttl_seconds=300
         )
         assert self.redis_connection.get(f"rate_limited_entity_{key_name}") is not None
+
+    def test_set_entity_to_rate_limited_error(self, mocker):
+        mock_redis = MagicMock(name="fake_redis")
+        mock_redis.set.side_effect = RedisError
+        key_name = "owner_id_123"
+        # This actually asserts that the error is not raised
+        # Despite the call failing
+        set_entity_to_rate_limited(
+            redis_connection=self.redis_connection, key_name=key_name, ttl_seconds=300
+        )
+        mock_redis.set.assert_not_called()
+
+    def test_set_entity_to_rate_limited_ttl_zero(self):
+        key_name = "owner_id_123"
+        # This actually asserts that the error is not raised
+        # Despite the call failing
+        set_entity_to_rate_limited(
+            redis_connection=self.redis_connection, key_name=key_name, ttl_seconds=0
+        )
+        assert self.redis_connection.get(f"rate_limited_entity_{key_name}") is not None
+
+    @pytest.mark.parametrize(
+        "app_id, installation_id",
+        [
+            (200, 718263),
+            (None, 718263),
+        ],
+    )
+    def test_gh_app_key_name_with_or_without_id(self, app_id, installation_id):
+        assert (
+            gh_app_key_name(
+                app_id=app_id,
+                installation_id=installation_id,
+            )
+            == f"{app_id}_{installation_id}"
+            if app_id
```

Comment:
[nit] We try to avoid if/else statements in test cases.

I think the clearer way to do this would be to have an extra parameter `expected`. And then you test simply

```
assert gh_app_key_name(app_id=app_id, installation_id=installation_id) == expected
```